### PR TITLE
Add Members History support

### LINKED ISSUE #2586: Create and maintain board member history table
## Summary

Create and maintain a `board_member_history` table, analogous to the `channelmemberhistory` table for Channels.

## How important this is to me and why

Importance: High

Use cases:
1. This is important for compliance and auditing

### PR DESCRIPTION
**Pull request by jespino**
Fixes  #2586

### CODE REVIEW

File: server/services/store/sqlstore/board.go [hunk 1]
```diff
@@ -528,3 +580,30 @@ func (s *SQLStore) searchBoardsForUserAndTeam(db sq.BaseRunner, term, userID, te
 
 	return s.boardsFromRows(rows)
 }
+
+func (s *SQLStore) getBoardMemberHistory(db sq.BaseRunner, boardID, userID string, limit uint64) ([]*model.BoardMemberHistoryEntry, error) {
```
Comment (jespino) [inline server/services/store/sqlstore/board.go:584]: This function and the `boardMemberHistoryEntriesFromRows` are just to be able to test it properly as part of the store, also, we are going to need something like that for the compliance export sooner than later, so I'm doing it anyway.

File: server/services/store/sqlstore/board.go [hunk 2]
```diff
@@ -384,6 +406,8 @@ func (s *SQLStore) saveMember(db sq.BaseRunner, bm *model.BoardMember) (*model.B
 		"scheme_viewer":    bm.SchemeViewer,
 	}
 
+	oldMember, _ := s.getMemberForBoard(db, bm.BoardID, bm.UserID)
```
Comment (jespino) [inline server/services/store/sqlstore/board.go:409]: This is not amazing, but is the simplest approach that I found to detect if this is an insert or an update.

File: server/services/store/sqlstore/board.go [hunk 3]
```diff
@@ -413,10 +448,27 @@ func (s *SQLStore) deleteMember(db sq.BaseRunner, boardID, userID string) error
...
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
 		return err
 	}
 
+	if rowsAffected > 0 {
+		addToMembersHistory := s.getQueryBuilder(db).
+			Insert(s.tablePrefix+"board_members_history").
+			Columns("board_id", "user_id", "action", "insert_at").
+			Values(boardID, userID, "delete", model.GetMillis())
```
Comment (chenilim) [inline server/services/store/sqlstore/board.go:465]: Let's make this "deleted" to be consistent with "created".

**Review by chenilim — CHANGES_REQUESTED** [review]
I've updated the spec to reflect this implementation. Please review. One requested change: "deleted" instead of "delete".

File: server/services/store/sqlstore/board.go (hunk 2 above)
Comment (jespino) [reply server/services/store/sqlstore/board.go:409]: fair point

Comment (jespino) [reply server/services/store/sqlstore/board.go]: Done.

**Review by chenilim — APPROVED** [review]
Looks good, thanks!

**Pull request merged**